src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1098,6 +1098,10 @@ def forward(self, inputs, targets):
                                ema_model=self.ema_model)
         self.phase_callback_handler(Phase.PRE_TRAINING, context)
 
+        first_batch, _ = next(iter(self.train_loader))
```

Comment:
We dont have the batch_size in this scope, so I get it manually. Do you see a nicer way to do ?